controllers/binding/snapshotenvironmentbinding_controller.go

```diff
@@ -107,6 +113,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 	adapter := NewAdapter(snapshotEnvironmentBinding, snapshot, environment, application, integrationTestScenario, logger, loader, r.Client, ctx)
 
+	// check for failed bindingconditions and attempt to clean up Ephemeral environments
+	// see STONEINTG-433
+	_, err = adapter.checkForBindingConditionsFailure()
```

Comment:
I think this would be better served by being an `Ensure*` function and called as part of the ReconcileHandler [here](https://github.com/redhat-appstudio/integration-service/pull/228/files#diff-5b178fa36b2385712f39d2575244cfca705af946a979640c73ed86f02e3ff0e0L111).